### PR TITLE
Fix failed CI by using more reasoning grep regex

### PR DESCRIPTION
**Pull request by unkcpz**

### CODE REVIEW

File: .docker/aiida-core-base/Dockerfile [hunk 1]
```diff
@@ -137,7 +137,7 @@ RUN set -x && \
         mamba && \
     rm micromamba && \
     # Pin major.minor version of python
-    mamba list python | grep '^python ' | tr -s ' ' | cut -d ' ' -f 1,2 >> "${CONDA_DIR}/conda-meta/pinned" && \
+    mamba list python | grep -oP 'python\s+\K[\d.]+' | tr -s ' ' | cut -d ' ' -f 1,2 >> "${CONDA_DIR}/conda-meta/pinned" && \
```
Comment (unkcpz) [inline .docker/aiida-core-base/Dockerfile:140]: This is the text from new `mamba list`

```
List of packages in environment: "/opt/conda"

  Name    Version  Build               Channel    
────────────────────────────────────────────────────
  python  3.10.13  hd12c33a_1_cpython  conda-forge
```

It added spaces indentation in front.

**Review by agoscinski — CHANGES_REQUESTED** [review]
Thanks! Can you add a comment to this regex expression? It is hard to digest. Otherwise I am fine.

File: .docker/aiida-core-base/Dockerfile (hunk 1 above)
Comment (unkcpz) [reply .docker/aiida-core-base/Dockerfile:140]: I was think about to add a comment with the text above. But I think it makes Dockerfile more confusing. 
The problem is the spaces indentation in front but the line is already indented. 
I'd rather not add these comment.

**Comment by unkcpz** [comment]
@agoscinski I think this is okay to merge? and also https://github.com/aiidateam/aiida-core/pull/6573 (I approved so please go ahead).

File: .docker/aiida-core-base/Dockerfile (hunk 1 above)
Comment (agoscinski) [reply .docker/aiida-core-base/Dockerfile:140]: Okay can you add this to the commit message at least?

**Review by agoscinski — APPROVED** [review]
Can you just put what you wrote in the commit mesage?

**Comment by agoscinski** [comment]
Maybe something like this:

The grep command failed that extracts the python version from `mamba list` failed because an indentation was added.
See the python string in the output of `mamba list`
```
List of packages in environment: "/opt/conda"

  Name    Version  Build               Channel    
────────────────────────────────────────────────────
  python  3.10.13  hd12c33a_1_cpython  conda-forge
```

File: .docker/aiida-core-base/Dockerfile (hunk 1 above)
Comment (danielhollas) [reply .docker/aiida-core-base/Dockerfile:140]: I think using awk for this (which ignores whitespace by default) would be much easier

```suggestion
    mamba list python | awk '$1 == "python" {print $2}' >> "${CONDA_DIR}/conda-meta/pinned" && \
```

Comment (unkcpz) [reply .docker/aiida-core-base/Dockerfile:140]: Sorry @danielhollas, I just saw your comment. If we remember next time when changing Dockerfile, will use `awk`. I did not familiar with awk :cry:

**Pull request merged**